scripts/taos-deploy-helper.sh

```
@@ -237,6 +237,238 @@ cmd_status() {
+        deps_snapshot="$("$venv/bin/pip" freeze 2>/dev/null || true)"
+        pkg_manager="pip"
+    fi
+
+    # Detect if uv is in use (uv.lock present in repo root)
+    if [[ -f "$repo_dir/uv.lock" ]]; then
+        pkg_manager="uv"
+        deps_snapshot="uv-lock:$(sha256sum "$repo_dir/uv.lock" 2>/dev/null | awk '{print $1}' || echo "unknown")"
+    fi
+
+    # Tag the current HEAD so it survives a checkout (detached or branch)
+    local tag="taos-worker-pre-update-$(date -u +%Y%m%d-%H%M%S)"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)